umap/umap_.py

```diff
@@ -2579,12 +2597,22 @@ def fit(self, X, y=None):
             print(ts(), "Construct embedding")
 
         if self.transform_mode == "embedding":
+            epochs = self.n_epochs_list if self.n_epochs_list is not None else self.n_epochs
             self.embedding_, aux_data = self._fit_embed_data(
                 self._raw_data[index],
-                self.n_epochs,
+                epochs,
                 init,
                 random_state,  # JH why raw data?
             )
+
+            if self.n_epochs_list is not None:
+                if "embedding_list" not in aux_data:
+                    raise KeyError("No list of embedding were found in 'aux_data'. "
+                                   "It is likely the layout optimization function "
+                                   "doesn't support the list of int for 'n_epochs'.")
+                else:
+                    self.embedding_list = [e[inverse] for e in aux_data["embedding_list"]]
```

Comment:
A Scikit-learn convention is that attributes set/created by a ``fit`` call should end in an underscore, so this should be ``self.embedding_list_``. Possibly you also want to use ``self.n_epochs_list_`` as well.

Reply:
Okay. But for `self.n_epochs_list`, it hasn't been created by fit(), it's just in the init, no?